### DISCUSSION TITLE
How to search movie by plot?

### DISCUSSION THREAD

**Discussion by manishh** [cinemagoer/cinemagoer#473] [Ideas]
If I need to search for a movie based on its plot summary, how can I do this using Cinemagoer?

**Comment by MiM-MiM** [comment]
I know this is old, but it doesn't look like they have this implemented.
You can search directly on the website to do this, it would be possible for cinemagoer to add these search fields.
https://www.imdb.com/search/title/?plot=Elphaba@@@%20the%20future%20Wicked%20Witch%20of%20the%20West%20and%20her%20relationship%20with%20Glinda@@@

That would work on the website.  It would be good to add all the advanced search options on that page.

If you need a python able one, I did create a python interface to the GraphQL API, which has this feature.
```python
from MyMovieGraphQL import Search
# A direct partial quote of a title plot.
Search.searchTitle(plotText="Elphaba, the future Wicked Witch of the West and her relationship with Glinda,")
# Or if you wanted titles that have a plot that contain both "Elphaba" and "Glinda"
Search.searchTitle(plotText=["Elphaba", "Glinda"], plotTextType="ALL")
# Can also do "ANY" for the type, allowing you to use it as searching for various terms. 
```
It is possible to do bot an ALL and ANY at the same time, I just didn't abstract that part in that search, but the module can still generate that search just fine.
https://github.com/MiM-MiM/MyMovieGraphQLPy

Or even if you need the json CLI version.
```
python -m MyMovieGraphQL searchTitle " " "plotText=Elphaba, the future Wicked Witch of the West and her relationship with Glinda,"
```
You can't do the two different plot texts in the CLI version currently, if people want that as an option it would be easy to add, just if the same key is found, make it a list and append rather than assume only one.

If you need help using this module just continue on the other repo. I only commented this here because the question was unanswered and this also provides another solution.

**Comment by alberanid** [comment]
This discussion is no longer relevant due to issue https://github.com/cinemagoer/cinemagoer/issues/557: as of June 2026 Cinemagoer no longer analyzes web pages and can only rely on non-commercial datasets.
